src/Lazy/dropRight.ts

```diff
@@ -0,0 +1,152 @@
+import isArray from "../isArray";
+import isString from "../isString";
+import toArray from "../toArray";
+import ReturnIterableIteratorType from "../types/ReturnIterableIteratorType";
+import { isAsyncIterable, isIterable } from "../_internal/utils";
+import concurrent, { isConcurrent } from "./concurrent";
+
+function* stringDropRight(length: number, str: string) {
+  const arr = Array.from(str);
+  for (let i = 0; i < str.length - length; i++) {
+    yield arr[i];
+  }
+}
+
+function* arrayDropRight<T>(length: number, arr: Array<T>) {
+  for (let i = 0; i < arr.length - length; i++) {
+    yield arr[i];
+  }
+}
+
+function* sync<T>(length: number, iterable: Iterable<T>) {
+  const arr = toArray(iterable);
+  for (let i = 0; i < arr.length - length; i++) {
+    yield arr[i];
+  }
+}
```

Comment:
```suggestion
function* stringDropRight(length: number, str: string) {
  for (let i = 0; i < str.length - length; i++) {
    yield str[i];
  }
}

function* sync<T>(length: number, iterable: Iterable<T>) {
  const arr = isArray(iterable) ? iterable : toArray(iterable);
  for (let i = 0; i < arr.length - length; i++) {
    yield arr[i];
  }
}
```

Reply:
It seems consistent to proceed with `Array.from` for string in the same way as `reverse`...!

```ts
Array.from('👍🤔')
['👍', '🤔']
```